device/transport/amqp/src/amqp.ts

```diff
@@ -92,7 +92,7 @@ export class Amqp extends EventEmitter implements Client.Transport {
     this._amqp = baseClient || new BaseAmqpClient(false, 'azure-iot-device/' + packageJson.version);
     this._amqp.setDisconnectHandler((err) => {
```

Comment:
>this._amqp.setDisconnectHandler [](start = 4, length = 31)

Why do we even have this mechanism.  Should this have just been a listener?

Reply:
historical reasons. yes. out of the scope of this PR to refactor though.

---
In reply to: [175831933](https://github.com/Azure/azure-iot-sdk-node/pull/254#discussion_r175831933) [](ancestors = 175831933)